webapp/channels/src/utils/desktop_app/auth.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import crypto from 'crypto';
+
+import {isDesktopApp} from 'utils/user_agent';
+
+export enum DesktopAuthStatus {
+    None,
+    Polling,
+    Expired,
+    Complete,
+}
+
+export const getExternalLoginURL = (url: string, search: string, token: string) => {
+    const params = new URLSearchParams(search);
+    if (isDesktopApp() && token) {
```

Comment:
0/5 but do we have to double check the user agent here, or is it it good enough to check that `token` is non-empty?

Reply:
I think the extra check is helpful to denote that this is Desktop App only, plus we shouldn't be sending these tokens with the webapp anyways.